aspnetcore/security/authentication/certauth.md

```diff
@@ -206,3 +206,243 @@ Complete the following steps In IIS Manager:
+            if (clientCertificate.Thumbprint == cert.Thumbprint)
+            {
+                return true;
+            }
+
+            return false;
+        }
+    }
+}
+```
+
+#### Implement an HttpClient using a certificate
+
+The web API client uses an `HttpClient` which was created using an `IHttpClientFactory` instance. This doesn't provide a way to define a handler for the `HttpClient`, so use an `HttpRequestMessage` to add the certificate to the `X-ARR-ClientCert` request header. The certificate is added as a string using the `GetRawCertDataString` method.
+ 
+```csharp
+private async Task<JsonDocument> GetApiDataAsync()
+{
+	try
+	{
+		var cert = new X509Certificate2(Path.Combine(_environment.ContentRootPath, "sts_dev_cert.pfx"), "1234");
+
+		var client = _clientFactory.CreateClient();
+
+		var request = new HttpRequestMessage()
+		{
+			RequestUri = new Uri("https://localhost:44379/api/values"),
+			Method = HttpMethod.Get,
+		};
+
+		request.Headers.Add("X-ARR-ClientCert", cert.GetRawCertDataString());
+		var response = await client.SendAsync(request);
+
+		if (response.IsSuccessStatusCode)
+		{
+			var responseContent = await response.Content.ReadAsStringAsync();
+			var data = JsonDocument.Parse(responseContent);
+
+			return data;
+		}
+
+		throw new ApplicationException($"Status code: {response.StatusCode}, Error: {response.ReasonPhrase}");
+	}
+	catch (Exception e)
+	{
+		throw new ApplicationException($"Exception {e}");
+	}
+}
+```
+
+If the correct certificate is sent to the server, the data is returned. If no certificate or the wrong certificate is sent, an HTTP 403 status code is returned.
+
+### Create certificates in PowerShell
+
+Creating the certificates is the hardest part in setting up this flow. A self-signed root certificate authority (CA) certificate is created using the `New-SelfSignedCertificate` PowerShell cmdlet. When creating the certificate, use a strong password. It's important to add the `KeyUsageProperty` parameter and the `KeyUsage` parameter as shown.
+
+#### Create root CA
+
+```powershell
+New-SelfSignedCertificate -DnsName "root_ca_dev_damienbod.com", "root_ca_dev_damienbod.com" -CertStoreLocation "cert:\LocalMachine\My" -NotAfter (Get-Date).AddYears(20) -FriendlyName "root_ca_dev_damienbod.com" -KeyUsageProperty All -KeyUsage CertSign, CRLSign, DigitalSignature
+
+$mypwd = ConvertTo-SecureString -String "1234" -Force -AsPlainText
+
+Get-ChildItem -Path cert:\localMachine\my\"The thumbprint..." | Export-PfxCertificate -FilePath C:\git\root_ca_dev_damienbod.pfx -Password $mypwd
+
+Export-Certificate -Cert cert:\localMachine\my\"The thumbprint..." -FilePath root_ca_dev_damienbod.crt
+```
+
+#### Install in the trusted root
+
+The root certificate needs to be trusted on your host system and a self-signed certificate will not be trusted per default. The following link explains how this could be done on a windows operating system:
```

Comment:
```suggestion
The root certificate needs to be trusted on your host system. A self-signed certificate won't be trusted by default. The following link explains how this can be accomplished on Windows:
```